File: android-app/app/src/main/java/com/tagtrack/app/core/domain/model/Item.kt
package com.tagtrack.app.core.domain.model

import java.time.Instant

data class Item(
    val id: String,
    val nfcUid: String,
    val name: String,
    val type: String,
    val color: String?,
    val brand: String?,
    val size: String?,
    val tags: List<String>,
    val notes: String?,
    val photoUri: String?,
    val photoCloudUrl: String?,
    val createdAt: Instant,
    val lastWorn: Instant?,
)

File: src/types/item.ts
export type ClothingType = 'Shirt' | 'Pants' | 'Shoes' | 'Accessory' | 'Outerwear' | 'Other';

export interface Item {
  id: string;
  name: string;
  type: ClothingType;
  color?: string;
  brand?: string;
  size?: string;
  tags: string[];
  notes?: string;
  photoURL?: string | null;
  nfcUID: string;
  createdAt: number;
  lastWorn?: number;
  deleted?: boolean;
}

export interface UserSettings {
  cloudBackup: boolean;
  analytics: boolean;
  lastBackupAt?: number;
}

export interface UserProfile {
  id: string;
  displayName?: string;
  email: string;
  settings: UserSettings;
}

export type ApiErrorCode =
  | 'NFCSCAN_DENIED'
  | 'NFCTAG_DUPLICATE'
  | 'ITEM_NOT_FOUND'
  | 'PERMISSION_DENIED'
  | 'NETWORK_ERROR';

export interface ApiError extends Error {
  code: ApiErrorCode;
}
